app/api/attachments/attachments.ts

```diff
@@ -34,12 +34,16 @@ export default {
       _id: { $ne: entity._id },
     });
 
-    const attachmentToDelete = (entity.attachments || []).find(a => a._id.equals(attachmentId));
+    const attachmentToDelete = (entity.attachments || []).find(
+      a => a._id && a._id.toString() === attachmentId.toString()
```

Comment:
@daneryl I'd like to understand why `a._id?.toString()` is not a better approach. As I understand it, it is, precisely, a shortcand to avoid these extra controls.  Also, it will return `undefined` if any of the checks fail, which is the same (mostly) as doing this booleans which would return false.  

So, using this in filters and other boolean comparators seems like the way to go.

Thanks for any feedback.